source/MaterialXContrib/MaterialXNode/MaterialXShadingNodeImpl.cpp

```diff
@@ -232,99 +232,53 @@ void MaterialXShadingNodeImpl<BASE>::updateShader(MHWRender::MShaderInstance& sh
-                status = shader.setParameter(resolvedName, floatVector);
-            }
-            else if (valueElement->getType() == MaterialX::TypedValue<MaterialX::Vector3>::TYPE)
-            {
-                MaterialX::Vector3 vector3 = valueElement->getValue()->asA<MaterialX::Vector3>();
-                MFloatVector floatVector(vector3[0], vector3[1], vector3[2]);
-                status = shader.setParameter(resolvedName, floatVector);
-            }
-            else if (valueElement->getType() == MaterialX::TypedValue<MaterialX::Vector4>::TYPE)
-            {
-                //				MaterialX::Vector4 vector4 = valueElement->getValue()->asA<MaterialX::Vector4>();
-                //				MFloatVector floatVector(vector4[0], vector4[1], vector4[2], vector4[3]);
-                //				status = shader.setParameter(resolvedName, floatVector);
-                //				std::cout << "updateShader (vector4): " << resolvedName << ". Status: " << status << std::endl;
-            }
-            else if (valueElement->getType() == MaterialX::TypedValue<MaterialX::Color2>::TYPE)
-            {
-                MaterialX::Color2 color2 = valueElement->getValue()->asA<MaterialX::Color2>();
-                MFloatVector floatVector(color2[0], color2[1]);
-                status = shader.setParameter(resolvedName, floatVector);
-            }
-            else if (valueElement->getType() == MaterialX::TypedValue<MaterialX::Color3>::TYPE)
-            {
-                MaterialX::Color3 color3 = valueElement->getValue()->asA<MaterialX::Color3>();
-                MFloatVector floatVector(color3[0], color3[1], color3[2]);
-                status = shader.setParameter(resolvedName, floatVector);
-            }
-            else if (valueElement->getType() == MaterialX::TypedValue<MaterialX::Color4>::TYPE)
-            {
-                //				MaterialX::Color4 color4 = valueElement->getValue()->asA<MaterialX::Color4>();
-                //                status = shader.setArrayParameter(resolvedName, color4.data(), 4);
-                //                std::cout << "updateShader (color4): " << resolvedName << std::endl;
-            }
-            else if (valueElement->getType() == MaterialX::TypedValue<MaterialX::Matrix44>::TYPE)
-            {
-                MaterialX::Matrix44 mat44 = valueElement->getValue()->asA<MaterialX::Matrix44>();
-                status = shader.setArrayParameter(resolvedName, mat44.data(), 16);
+                // TODO: This should come from the element and not hard-coded.
+                MHWRender::MSamplerStateDesc samplerDescription;
+                samplerDescription.filter = MHWRender::MSamplerState::kAnisotropic;
+                samplerDescription.maxAnisotropy = 16;
+
+                status = bindFileTexture(shader, textureParameterName, imageSearchPath, valueString,
+                    samplerDescription, textureDescription);
             }
         }
     }
 }
 
-class MaterialXTextureOverride;
```

Comment:
These forward declarations were once necessary because the template specialized on these types, not anymore.